### ISSUE TITLE
Android/Quest suboptimal swapchain error

### ISSUE THREAD

**Issue by BastiaanOlij** [godotengine/godot#74963]
### Godot version

4.0.stabl

### System information

Android - Meta Quest 2

### Issue description

When running a project on a Quest 2 using the Vulkan renderer (I believe there is no difference between mobile and clustered Vulkan renderer) the log gets filled with these repeated errors:

```
Vulkan: Early out of date swapchain, recreating.
Vulkan queue submit: Swapchain is suboptimal.
Vulkan: Early out of date swapchain, recreating.
Vulkan queue submit: Swapchain is suboptimal.
```

While discovered on the Quest this issue may not be limited to this device.

Need to investigate the origin of these errors. My current theory is that this may be the swap chain the renderer creates by default for outputting to screen. This swap chain is unused (as when using XR the XR Runtime is responsible for output to the device) but still maintained.

### Steps to reproduce

Run a project on a Meta Quest (2)

### Minimal reproduction project

The demo in XR tools can be used for this: https://github.com/GodotVR/godot-xr-tools

**Comment by BastiaanOlij** [comment]
@szymonm-google , Fredia asked me to tag you in this, possibly something the Android team could shine a spotlight on?

@rbessems , is this familiar to your team? At face value this may be caused by the default swapchain we're creating as part of the display server, for which we're working on #74754. I haven't had time to continue on that as I'm focused on GDC atm.

**Comment by szymonm-google** [comment]
Hey Bas! Sure let me take a look

**Comment by zhangyiwei** [comment]
This would help: https://android-developers.googleblog.com/2020/02/handling-device-orientation-efficiently.html

Vulkan game engine on Android mobile platform needs to handle device pre-rotation explicitly for the optimal and stable performance.

**Comment by BastiaanOlij** [comment]
> This would help: https://android-developers.googleblog.com/2020/02/handling-device-orientation-efficiently.html
> 
> Vulkan game engine on Android mobile platform needs to handle device pre-rotation explicitly for the optimal and stable performance.

That would underline my theory that this is due to the default swap chain we're creating and that really shouldn't be used anymore in an XR environment. The device rotation is no longer a factor when we're dealing with a Quest or other XR device.

**Comment by BastiaanOlij** [comment]
Just copying in a reaction from @clayjohn on the Contributors chat

> I've already made a PR to fix that godot#72859 Applications are allowed to continue to operate with suboptimal swapchains, but there can be performance implications, so it's best to just recreate in such cases

https://github.com/godotengine/godot/pull/72859
